app/server/lib/Authorizer.ts

```diff
@@ -193,6 +193,24 @@ export async function addRequestUser(
     }
   }
 
+  // Check if we have a boot key. This is a fallback mechanism for an
+  // administrator to authenticate themselves by demonstrating access
+  // to the environment.
+  if (!authDone && mreq.headers && mreq.headers['x-boot-key']) {
```

Comment:
@dsagal can I ask you to look over this section, as a sanity-check for security. If there were a bug, like a flipped test, I could imagine this leaving a gaping hole.